great_expectations/data_context/types/base.py

```diff
@@ -2084,8 +2084,9 @@ def __init__(
             plugins_directory = (
                 DataContextConfigDefaults.DEFAULT_PLUGINS_DIRECTORY.value
             )
-        self.plugins_directory = plugins_directory
+        self.plugins_directory = str(plugins_directory)
         if root_directory is not None:
+            root_directory = str(root_directory)
```

Comment:
I discovered that this class only accepts strings instead of `pathlib.Path` objects or strings.
This is the only core code change in this PR.